### PR TITLE
Plug

### PR DESCRIPTION
**Pull request by uk4z**

### CODE REVIEW

File: src/lib/state.ts [hunk 1]
```diff
@@ -113,6 +124,43 @@ export async function fetchBalances(
...
+const agent = Promise.resolve(new HttpAgent({ host: HOST, identity })).then(async (agent) => {
+	await agent.fetchRootKey();
+	agent.addTransform('query', makeNonceTransform());
+	return agent;
+});
+
+async function createSecp256k1IdentityActor(
+	canisterId: Principal,
+	idl: IDL.InterfaceFactory,
+	seed?: number
```
Comment (leo-ship-it) [inline src/lib/state.ts:139]: why? you never use this

File: src/lib/state.ts [hunk 2]
```diff
@@ -113,6 +124,43 @@ export async function fetchBalances(
...
+): Promise<any> {
+	let seed1: Uint8Array | undefined;
+	if (seed) {
+		seed1 = new Uint8Array(new Array(32).fill(0));
+		seed1[0] = seed;
+	}
```
Comment (leo-ship-it) [inline src/lib/state.ts:145]:
```suggestion
seed1 = new Uint8Array(new Array(32).fill(0));
```

File: src/lib/state.ts [hunk 3]
```diff
@@ -113,6 +124,43 @@ export async function fetchBalances(
...
+
+	const identity = Ed25519KeyIdentity.generate(seed1);
+	const agent1 = new HttpAgent({ source: await agent, identity });
```
Comment (leo-ship-it) [inline src/lib/state.ts:148]:
```suggestion
	const dummyAgent = new HttpAgent({ source: await agent, identity });
```

File: src/lib/state.ts (hunk 1 above)
Comment (uk4z) [reply src/lib/state.ts]:
```
async wtnAllocation(principal: Principal): Promise<BigNumber | undefined> {
		try {
			const waterNeuron = await createSecp256k1IdentityActor(
				Principal.fromText(CANISTER_ID_WATER_NEURON),
				idlFactoryWaterNeuron
			);
			const allocation = await waterNeuron.get_airdrop_allocation([principal]);
			return bigintE8sToNumber(allocation);
		} catch (e) {
			console.log('Error while fetching airdrop allocation:', e);
		}
	}
	
	```

File: src/lib/state.ts (hunk 2 above)
Comment (uk4z) [reply src/lib/state.ts]: Yeah I changed the code completely in the new preview

File: src/lib/state.ts (hunk 3 above)
Comment (uk4z) [reply src/lib/state.ts]: good comment. Done.

**Pull request merged**